Packet++/header/TcpReassembly.h

```diff
@@ -301,6 +315,9 @@ class TcpReassembly
 	 */
 	void reassemblePacket(RawPacket* tcpRawData);
 
+	void reassemblePacket(RawPacket* tcpRawData, ReassemblyStatus &status);
+	void reassemblePacket(Packet& tcpData, ReassemblyStatus &status);

```

Comment:
I'm wondering whether we should provide a different API with the status or just add this to the current API.

On one hand if we add it to the current API it won't be backward compatible, on the other hand it's quite confusing to have so many methods doing the same thing.

I'm leaning towards adding the status to the existing API, please let me know what you think

Reply:
I agree with you. I removed the redundant method declarations in the new commit. I also touched some test and example code to make them compile.

Reply:
Thank you!

I have another idea that I'd like to share with you: what if instead of adding the status as a second parameter to `reassemblePacket()` we can just let it return this enum value?

That way the API change will be backward compatible and we still have this feature added.

Also, it'd be great to add a few test cases in `Pcap++Test` but I can do it for you after the PR is merged.

Please let me know what you think

Reply:
Way better than my approach. Implemented in [1953f80](https://github.com/seladb/PcapPlusPlus/pull/435/commits/1953f803ddadb935bdd248d92f1d2a67d117cea6)